src/test/java/gil/server/RouterTest.java

```diff
@@ -0,0 +1,27 @@
+package gil.server;
+
+import org.junit.Test;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import java.io.UnsupportedEncodingException;
+
+public class RouterTest {
+    Router router = new Router();
+
+    @Test
+    public void shouldReturnTrueIfEndpointExists() {
+        assertTrue(router.hasRoute("/"));
+    }
+
+    @Test
+    public void shouldReturnFalseIfEndpointDoesNotExist() {
+        assertFalse(router.hasRoute("/fake_endpoint"));
+    }
+
+    @Test
+    public void shouldReturnAResponseIfTheRouteExists() throws UnsupportedEncodingException {
```

Comment:
Similar comment here as the one left in `HandlerTest` -> what should `Response` actually look like here? 
Also, it would be good to test the "sad path" of the `route` method -> what happens if the route does not exist?